### DISCUSSION TITLE
Support for discrete tabular and text models

### DISCUSSION THREAD

**Discussion by NeelimaRGade** [Trusted-AI/adversarial-robustness-toolbox#2236] [Ideas]
**Is your feature request related to a problem? Please describe.**
ART metrics can't be computed for discrete tabular data. I don't see any sample notebooks for computing ART metrics for text models as well.
**Describe the solution you'd like**
For discrete tabular models, ART should convert the features internally into one hot encoded data and support generating the metrics.

cc: @beat-buesser, @kieranfraser

**Comment by beat-buesser** [comment]
Hi @NeelimaRGade Thank you very much for proposing this feature. Do you have a specific application example where you could share a code example?

**Comment by beat-buesser** [comment]
I convert this issue into a Discussion to facilitate generating specific feature issues.

**Comment by NeelimaRGade** [comment]
Hi @beat-buesser, please find attached the notebook I used to generate ART metrics for German Credit Risk data. 


[SHAPr_tabular_data copy.ipynb.txt](https://github.com/Trusted-AI/adversarial-robustness-toolbox/files/12287460/SHAPr_tabular_data.copy.ipynb.txt)
